src/main/java/com/github/lordrex34/config/model/ConfigFieldInfo.java

```diff
@@ -0,0 +1,204 @@
+public final class ConfigFieldInfo
+{
+	private static final Logger LOGGER = LoggerFactory.getLogger(ConfigFieldInfo.class);
+	
+	/** To avoid creating the same value supplier thousand times. */
+	private static final Map<String, IConfigValueSupplier<?>> VALUE_SUPPLIERS = new HashMap<>();
+	
+	/** To avoid creating the same post load hook thousand times. */
+	private static final Map<String, ConfigPostLoadHook> POST_LOAD_HOOKS = new HashMap<>();
+	
+	/** The class that is being scanned. */
+	private final Class<?> _clazz;
+	
+	/** The field that contains {@link ConfigField} annotation. */
+	private final Field _field;
+	
+	/** The configuration field information annotation. */
+	private final ConfigField _configField;
+	
+	/** Group beginning marker annotation. */
+	private final ConfigGroupBeginning _beginningGroup;
+	
+	/** Group ending marker annotation. */
+	private final ConfigGroupEnding _endingGroup;
+	
+	/**
+	 * Constructs a new information container class for the field.
+	 * @param clazz the class that is being scanned
+	 * @param field the field that contains {@link ConfigField} annotation
+	 */
+	public ConfigFieldInfo(Class<?> clazz, Field field)
+	{
+		_clazz = clazz;
+		_field = field;
+		_configField = _field.getDeclaredAnnotation(ConfigField.class);
+		_beginningGroup = _field.getDeclaredAnnotation(ConfigGroupBeginning.class);
+		_endingGroup = _field.getDeclaredAnnotation(ConfigGroupEnding.class);
+	}
+	
+	/**
+	 * Loads and configures the field with its proper values.
+	 * @param configPath path of the properties file
+	 * @param properties the regular properties
+	 * @param overriddenProperties the content of {@code override.properties}
+	 */
+	public void load(Path configPath, PropertiesParser properties, PropertiesParser overriddenProperties)
+	{
+		// Skip inappropriate fields.
+		if (!Modifier.isStatic(_field.getModifiers()) || Modifier.isFinal(_field.getModifiers()))
+		{
+			LOGGER.debug("Skipping non static or final field: {}#{}", _clazz.getSimpleName(), _field.getName());
+			return;
+		}
+		
+		// If field is just a comment holder, then do not try to load it.
+		if (_configField.onlyComment())
+		{
+			return;
+		}
+		
+		try
+		{
+			final String propertyKey = _configField.name();
+			ConfigManager.getInstance().registerProperty(_clazz.getPackage().getName(), configPath, propertyKey);
+			if (!_configField.reloadable() && ConfigManager.getInstance().isReloading())
+			{
+				LOGGER.debug("Property '{}' retained with its previous value!", propertyKey);
+				return;
+			}
+			
+			// private field support
+			final boolean wasAccessible = _field.isAccessible();
+			if (!wasAccessible)
+			{
+				_field.setAccessible(true);
+			}
+			
+			// value suppliers
+			final Class<? extends IConfigValueSupplier<?>> valueSupplierClass = _configField.valueSupplier();
+			final String valueSupplierClassName = valueSupplierClass.getName();
+			IConfigValueSupplier<?> valueSupplier = VALUE_SUPPLIERS.get(valueSupplierClassName);
```

Comment:
Come on... computeIfAbsent pls. There are other places like that as well.

Reply:
```
			// value suppliers
			final Class<? extends IConfigValueSupplier<?>> valueSupplierClass = _configField.valueSupplier();
			final String valueSupplierClassName = valueSupplierClass.getName();
			IConfigValueSupplier<?> valueSupplier = VALUE_SUPPLIERS.computeIfAbsent(valueSupplierClassName, k ->
			{
				try
				{
					return valueSupplierClass.newInstance();
				}
				catch (InstantiationException | IllegalAccessException e)
				{
					e.printStackTrace();
					return null;
				}
			});
```


Feels much cleaner, huh? xD